cli/azd/cmd/middleware/telemetry.go

```diff
@@ -62,6 +69,18 @@ func (m *TelemetryMiddleware) Run(ctx context.Context, next NextFn) (*actions.Ac
 
 	span.SetAttributes(fields.CmdArgsCount.Int(len(m.options.Args)))
 
+	// Track all alpha features that are enabled for the current command
+	if allFeatures, err := m.alphaFeatureManager.ListFeatures(); err == nil && len(allFeatures) > 0 {
```

Comment:
With the current approach, we would always do upfront calculation for feature flags for telemetry purposes. This does introduce a consistent performance overhead on all commands.

One thing that's stuck with me when working on telemetry, and a goal I have, is to try to make it minimally costly for users as well. For example, in [project.go](https://github.com/Azure/azure-dev/blob/a17935e47211ed01780281974c6d63b467de5b14/cli/azd/pkg/project/project.go#L136), we only emit telemetry for project config since we needed to load it anyways. The in-memory computations are extremely cheap compared to forcing on-disk load.

I was thinking if we would want to do something similar here, where we would do `tracing.SetUsageAttribute` on the result of [IsEnabled](https://github.com/Azure/azure-dev/blob/a17935e47211ed01780281974c6d63b467de5b14/cli/azd/pkg/alpha/alpha_feature_manager.go#L76). 

I'm thinking we'd want to track the unique set of feature flags, and I can make some changes in our attribute collection to make that easier if all this sounds good to you.